<robot name="cup">
  <link name="base_link"/>

  <joint name="cylinder_fixed_joint" type="fixed">
    <parent link="base_link"/>
    <child link="cylinder"/>
    <origin xyz="0 0 0.5"/>
  </joint>

  <link name="cylinder">
    <inertial>
      <origin xyz="0.0 0.0 0.0" rpy="0 0 0" />
      <mass value="1.2"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0"
	       iyy="0.001" iyz="0.0"
	       izz="0.001" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://agimus_demos/meshes/cylinder.stl"/>
      </geometry>
      <material name="lightGrey">
        <color rgba="0.9 0.9 0.9 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://agimus_demos/meshes/cylinder.stl"/>
      </geometry>
    </collision>
  </link>

  <link name="tag36_11_00100">
    <visual>
      <origin xyz="0 0 0.00001"/>
      <geometry>
        <mesh filename="package://gerard_bauzil/meshes/apriltag_36h11/tag36_11_00100.dae" scale="0.0573 0.0573 1."/>
      </geometry>
    </visual>
  </link>
  <joint name="to_tag100" type="fixed">
    <parent link="cylinder"/>
    <child link="tag36_11_00100"/>
    <origin xyz="0.5 0 0" rpy="0 1.5707963267948966 0"/>
  </joint>
  <link name="tag36_11_00101">
    <visual>
      <origin xyz="0 0 0.00001"/>
      <geometry>
        <mesh filename="package://gerard_bauzil/meshes/apriltag_36h11/tag36_11_00101.dae" scale="0.0573 0.0573 1."/>
      </geometry>
    </visual>
  </link>
  <joint name="to_tag101" type="fixed">
    <parent link="cylinder"/>
    <child link="tag36_11_00101"/>
    <origin xyz="-0.5 0 0" rpy="0 -1.5707963267948966 0"/>
  </joint>

  <link name="tag36_11_00102">
    <visual>
      <origin xyz="0 0 0.00001"/>
      <geometry>
        <mesh filename="package://gerard_bauzil/meshes/apriltag_36h11/tag36_11_00102.dae" scale="0.0573 0.0573 1."/>
      </geometry>
    </visual>
  </link>
  <joint name="to_tag102" type="fixed">
    <parent link="cylinder"/>
    <child link="tag36_11_00102"/>
    <origin xyz="0 0.5 0" rpy="-1.5707963267948966 0 0"/>
  </joint>
  <link name="tag36_11_00103">
    <visual>
      <origin xyz="0 0 0.00001"/>
      <geometry>
        <mesh filename="package://gerard_bauzil/meshes/apriltag_36h11/tag36_11_00103.dae" scale="0.0573 0.0573 1."/>
      </geometry>
    </visual>
  </link>
  <joint name="to_tag103" type="fixed">
    <parent link="cylinder"/>
    <child link="tag36_11_00103"/>
    <origin xyz="0 -0.5 0" rpy="1.5707963267948966 0 0"/>
  </joint>

  <gazebo>
    <static>true</static>
  </gazebo>
</robot>
